### PR TITLE
Rename generic run-on flag into platform

### LINKED ISSUE #6318: Rename generic `run-on` flag into `platform`
As an odo user, I want the experimental generic `--run-on` flag to be renamed into `--platform` because it makes more sense for listing, deleting, or describing components.

## Acceptance Criteria
- [ ] `--run-on` should be renamed into `--platform` everywhere (help output, ...)
- [ ] Internal code (e.g., variables names) should reflect that change as well
- [ ] `--platform` should be experimental
- [ ] `--platform` should be documented in the [list of experimental features](https://odo.dev/docs/user-guides/advanced/experimental-mode#list-of-experimental-features)

## Related Epic
https://github.com/redhat-developer/odo/issues/5871

Needed for the following issues:
- https://github.com/redhat-developer/odo/issues/6296
- https://github.com/redhat-developer/odo/issues/6297
- https://github.com/redhat-developer/odo/issues/6298

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind feature

**What does this PR do / why we need it:**

- rename `--run-on` flag with `--platform`
- change "RUNNING ON" header with "PLATFORM" in output of `odo list component`
- duplicate `runningOn` with `platform` field in JSON output of `odo list component` and deprecate `runningOn`

**Which issue(s) this PR fixes:**

Fixes #6318 

**PR acceptance criteria:**

- [x] Unit test 

- [x] Integration test 

- [x] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: docs/website/blog/2022-11-29-odo-v3.3.0.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ Learn more about the experimental mode in [this doc](../docs/user-guides/advance
 <iframe width="560" height="315" src="https://www.youtube.com/embed/EmN_hGkOHX4?list=PLGMB2PY4SNOr5pjbPTdcWsIrOezCPXWdX" title="odo dev using podman" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
 
 You can now use`odo dev` to run your application on Podman. `odo` leverages the benefit of Podman to work seamlessly with Kubernetes YAML and replicate them to containers locally.
-This is an experimental feature and can be used with `--run-on` flag.
+This is an experimental feature and can be used with `--platform` flag.
```
Comment (rm3l) [inline docs/website/blog/2022-11-29-odo-v3.3.0.md:34]: I think we should keep `run-on` in this blog post, as it is specific and relevant to `3.3.0`

File: pkg/api/component-abstract.go [hunk 2]
```diff
@@ -9,7 +9,10 @@ type ComponentAbstract struct {
 	RunningIn RunningModes `json:"runningIn"`
 	Type      string       `json:"projectType"`
 	// RunningOn is the platform the component is running on, either cluster or podman
+	// This field is deprecated and will be replaced by Platform
 	RunningOn string `json:"runningOn,omitempty"`
```
Comment (rm3l) [inline pkg/api/component-abstract.go:13]: Can you also mark this field as deprecated with the `// Deprecated: ` comment?

File: pkg/odo/cli/feature/features.go [hunk 3]
```diff
@@ -13,8 +13,8 @@ type OdoFeature struct {
 }
 
 var (
-	// GenericRunOnFlag is the feature supporting the `--run-on` generic CLI flag.
-	GenericRunOnFlag = OdoFeature{
+	// GenericPformFlag is the feature supporting the `--platform` generic CLI flag.
+	GenericPformFlag = OdoFeature{
```
Comment (rm3l) [inline pkg/odo/cli/feature/features.go:17]: A few characters missing here - I think you meant `GenericPlatformFlag` ;-)

**Review by rm3l — CHANGES_REQUESTED** [review]
Added a few comments. I also noticed that there were a few changes related to #6437 (`odo delete component --name` on Podman). Can you rebase this branch onto `main` instead?

**Review by rm3l — COMMENTED** [review]
One last comment. Can you also update the following occurrences accordingly?

```
$ git grep 'run-on' -- ':!docs/website/blog' ':!vendor'

tests/integration/cmd_devfile_list_test.go:                     It("should display component depending on experimental mode and run-on flag", func() {
tests/integration/cmd_devfile_list_test.go:                                     By("returning component not in dev mode when experimental mode is enabled with json output and run-on is cluster", func() {
```

**Comment by feloy** [comment]
Closed in favor of https://github.com/redhat-developer/odo/pull/6445 (based on main instead of branch)

**Pull request closed without merging**